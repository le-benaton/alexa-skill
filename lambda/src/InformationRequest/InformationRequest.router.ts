import { Router } from "@talkyjs/core";
import { getSlotValue } from 'ask-sdk-core';

import axios from 'axios';
import { IData, getInformation, IInformationType, InformationRequestScript } from './InformationRequest.speech';

export const InformationRequestRouter: Router = {
    requestType: "IntentRequest",
    intentName: "InformationIntent",

    handler: async (handlerInput) => {
        const informationType = getSlotValue(handlerInput.requestEnvelope, 'InformationType') as IInformationType;
        console.log({
            informationType,
            request: handlerInput.requestEnvelope,
        });
        const { data } = await getInformation();

        const script = new InformationRequestScript(handlerInput)

        script.setData(data as IData[], informationType);
        return script
            .createResponseBuilder()
            .withShouldEndSession(true)
            .getResponse();
    }
}

export default InformationRequestRouter
