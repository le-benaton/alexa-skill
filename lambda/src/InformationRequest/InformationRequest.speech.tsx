/** @jsx ssml */
import {
    ssml,
    SpeechScriptJSX,
} from '@ask-utils/speech-script'
import axios from 'axios';

export interface IData {
    title: string;
    body: string;
}

export type IInformationType = 'NEWS' | 'HOLIDAY'

export const getInformation = async () => {
    const data = await axios.get('https://www.benaton.net/api/news.json');
    return data;
}

const holiday = [
  "おやすみ",
  "やすみ",
  "定休日"
];

export class InformationRequestScript extends SpeechScriptJSX {
    private data: string = '';

    setData(data: IData[], informationType: IInformationType) {
        this.data = data
          .filter(item => {
              /**
               * ベナトンの更新者が title につける値
               * https://www.benaton.net/
               */
              const isHoliday = item.title.includes('おやすみ') || item.title.includes('定休日') || item.title.includes('お休み');
              return holiday.includes(informationType) ? isHoliday : !isHoliday;
          })
          .map(item => {
            return `「${item.title.replace(/<("[^"]*"|'[^']*'|[^'">])*>/g,'')}」についてご案内します。${item.body.replace(/<("[^"]*"|'[^']*'|[^'">])*>/g,'')}`
        }).join('')
    }

    speech() {
        return (
            <speak><p>{this.data}</p></speak>
        )
    }

    reprompt() {
        return (
          <speak><p>{this.data}</p></speak>
        )
    }

}
