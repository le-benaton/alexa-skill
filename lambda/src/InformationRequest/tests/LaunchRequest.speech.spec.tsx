import { HandlerInputCreator } from '@ask-utils/test';
import { InformationRequestScript } from '../InformationRequest.speech'

describe('InformationRequestRouter', () => {

  it('should return false when given a not IntentRequest', async () => {
    const handlerInput = new HandlerInputCreator().createIntentRequest({
      name: "IntentRequest",
      confirmationStatus: 'NONE'
    });
    const script = new InformationRequestScript(handlerInput);
    expect(script.createResponse()).toMatchSnapshot();
  });

});
