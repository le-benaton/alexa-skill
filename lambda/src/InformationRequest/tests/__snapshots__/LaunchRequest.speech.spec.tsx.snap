// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InformationRequestRouter should return false when given a not IntentRequest 1`] = `
Object {
  "outputSpeech": Object {
    "ssml": "<speak><p/></speak>",
    "type": "SSML",
  },
  "reprompt": Object {
    "outputSpeech": Object {
      "ssml": "<speak><p/></speak>",
      "type": "SSML",
    },
  },
  "shouldEndSession": false,
}
`;

exports[`LaunchRequestScript should return false when given a not IntentRequest 1`] = `
Object {
  "outputSpeech": Object {
    "ssml": "<speak><p>Hello! It&apos;s a nice development. How are you?</p></speak>",
    "type": "SSML",
  },
  "reprompt": Object {
    "outputSpeech": Object {
      "ssml": "<speak><p>How are you?</p></speak>",
      "type": "SSML",
    },
  },
  "shouldEndSession": false,
}
`;
